app/src/App.tsx

```diff
@@ -1,48 +1,77 @@
 import React from 'react'
-import Web3Provider from 'web3-react'
+import Web3Provider, { useWeb3Context, Web3Consumer } from 'web3-react'
 import WalletConnectQRCodeModal from '@walletconnect/qrcode-modal'
-import { WalletConnect } from './connectors'
-import { useWeb3Context } from 'web3-react'
+import connectors from './connectors'
 
 const ConnectWallet: React.FC = () => {
   const context = useWeb3Context()
 
-  const connect = () => {
-    context.setConnector('WalletConnect', { networkId: 4 })
+  if (context.error) {
+    console.error('Error in web3 context', context.error)
   }
 
   React.useEffect(() => {
-    if (context.active && !context.account) {
+    if (context.active && !context.account && context.connectorName === 'WalletConnect') {
       const uri = context.connector.walletConnector.uri
       WalletConnectQRCodeModal.open(uri, () => {})
 
       context.connector.walletConnector.on('connect', () => {
         WalletConnectQRCodeModal.close()
       })
+    } else {
+      try {
+        WalletConnectQRCodeModal.close()
+      } catch (err) {
+        console.error(err)
+      }
     }
   }, [context, context.active])
 
-  return <button onClick={connect}>Connect</button>
-}
-
-const Account: React.FC = () => {
-  const context = useWeb3Context()
+  return (
+    <>
+      {Object.keys(connectors).map(connectorName => (
```

Comment:
PP: Unnecessarily generic at this point. Plus, if we are going to support only one wallet for now, then just a `Connect Wallet` button should be enough.

Reply:
Done!